Apps/Src/Temperature.c

```diff
@@ -26,12 +30,25 @@ ErrorStatus Temperature_Init(cell_asic *boards){
 	wakeup_sleep(NUM_MINIONS);
 	LTC6811_Init(Minions);
 
+	//take control of mutex
+	OS_ERR err;
+  	OSMutexPend(&MinionsASIC_Mutex, 0, OS_OPT_PEND_BLOCKING, NULL, &err);
+  	assertOSError(err);
```

Comment:
@SijWoo do you know if this is possible.

Also, what are the issues you are referring to @chaseblock ? The `assertOSError(err)` should put us into a fault state if an error is returned, which would mean that we don't read the minions array anymore.

Reply:
Do we plan on reading sensor data in the fault condition (or at least trying)?

I suppose that since a fault should be the only thread running at that point, it's unreasonable to be checking mutexes. I was just talking about the possibility of entering some state where a mutex is held by some task that has been killed. Probably not actually a concern on second thought.